wrapper/Ada/README.md

```diff
@@ -0,0 +1,110 @@
+avoiding the secondary stack using the restriction
+No_Secondary_Stack (see the GNAT configuration file gnat.adc
+which instructs compilation of the WolfSSL Ada binding under
+this restriction).
+
+Portability: The WolfSSL Ada binding makes no usage of controlled types
+and has no dependency upon the Ada.Finalization package.
+Lighter Ada run-times for embedded systems often have
+the restriction No_Finalization. The WolfSSL Ada binding has
+been developed with maximum portability in mind.
+
+Not only can the WolfSSL Ada binding be used in Ada applications but
+also SPARK applications (a subset of the Ada language suitable
+formal verification). To formally verify the Ada code in this repository
+open the client.gpr with GNAT Studio and then select
+SPARK -> Prove All Sources and use Proof Level 2.
+
+Summary of SPARK analysis
+=========================
+
+---------------------------------------------------------------------------------------------------------------
+SPARK Analysis results        Total        Flow   CodePeer                       Provers   Justified   Unproved
+---------------------------------------------------------------------------------------------------------------
+Data Dependencies                 2           2          .                             .           .          .
+Flow Dependencies                 .           .          .                             .           .          .
+Initialization                   15          15          .                             .           .          .
+Non-Aliasing                      .           .          .                             .           .          .
+Run-time Checks                  58           .          .    58 (CVC4 85%, Trivial 15%)           .          .
+Assertions                        6           .          .                      6 (CVC4)           .          .
+Functional Contracts             91           .          .                     91 (CVC4)           .          .
+LSP Verification                  .           .          .                             .           .          .
+Termination                       .           .          .                             .           .          .
+Concurrency                       .           .          .                             .           .          .
+---------------------------------------------------------------------------------------------------------------
+Total                           172    17 (10%)          .                     155 (90%)           .          .
+
+## Compiler and Build System installation
+
+### GNAT Community Edition 2021
+Download and install the GNAT community Edition 2021 compiler and studio:
+https://www.adacore.com/download
+
+Linux Install:
+
+```sh
+chmod +x gnat-2021-20210519-x86_64-linux-bin
+./gnat-2021-20210519-x86_64-linux-bin
+```
+
+```sh
+export PATH="/opt/GNAT/2021/bin:$PATH"
+gprclean
+gprbuild default.gpr
+gprbuild client.gpr
```

Comment:
The command fails for me with :

```
   [C]            quic.c
   [C]            t.c
/home/jak/Documents/wolfssl/t.c:6:5: warning: no previous prototype for 'test_1' [-Wmissing-prototypes]
    6 | int test_1(void)
      |     ^~~~~~
/home/jak/Documents/wolfssl/t.c:11:6: warning: no previous prototype for 'test_2' [-Wmissing-prototypes]
   11 | void test_2(int a, int b)
      |      ^~~~~~
/home/jak/Documents/wolfssl/t.c: In function 'test_2':
/home/jak/Documents/wolfssl/t.c:11:24: warning: unused parameter 'b' [-Wunused-parameter]
   11 | void test_2(int a, int b)
      |                    ~~~~^
/home/jak/Documents/wolfssl/t.c: At top level:
/home/jak/Documents/wolfssl/t.c:16:5: warning: function declaration isn't a prototype [-Wstrict-prototypes]
   16 | int main()
      |     ^~~~
/home/jak/Documents/wolfssl/t.c: In function 'main':
/home/jak/Documents/wolfssl/t.c:4:19: error: expected ')' before ';' token
    4 | #define Y test_1(); X
      |                   ^
/home/jak/Documents/wolfssl/t.c:2:28: note: in definition of macro 'test'
    2 | #define test(a, b) test_2((a), b)
      |                            ^
/home/jak/Documents/wolfssl/t.c:18:10: note: in expansion of macro 'Y'
   18 |     test(Y, X);
      |          ^
/home/jak/Documents/wolfssl/t.c:2:27: note: to match this '('
    2 | #define test(a, b) test_2((a), b)
      |                           ^
/home/jak/Documents/wolfssl/t.c:18:5: note: in expansion of macro 'test'
   18 |     test(Y, X);
      |     ^~~~
gprbuild: *** compilation phase failed

```

Am I missing a step?

Reply:
This is from me not starting with a clean directory when trying to build, it picked up other source files.

Reply:
That happened to me too. It might be good to put a note in the readme.